sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -20,6 +20,146 @@
 s3 = boto3.client("s3")
 
 
+def s3_event_handler(event, engine):
+    """Handler function for S3 events.
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+    engine : sqlalchemy.engine.Engine
+        Engine object to interact with the database.
+
+    Returns
+    -------
+    dict
+        HTTP response
+    """
+    # Retrieve the Object name
+    s3_filepath = event["detail"]["object"]["key"]
+
+    filename = os.path.basename(s3_filepath)
+    # TODO: add checks for SPICE or other
+    # data types
+    science_file = ScienceFile(filename)
+
+    if science_file.is_valid is False:
+        return {"statusCode": 400, "body": science_file.error_message}
+
+    # setup a dictionary of metadata parameters to unpack in the
+    # file catalog table
+    metadata_params = science_file.get_file_metadata_params()
+    metadata_params["file_path"] = s3_filepath
+
+    if metadata_params["data_level"] == "l0":
+        logger.info(f"Inserting {filename} into database")
+        # Add data to the file catalog and status tables
+        with Session(engine) as session:
+            session.add(models.FileCatalog(**metadata_params))
+            # Event time looks like this:
+            # "time": "2024-01-16T17:35:08Z"
+
+            # Parses the time string from the event to a datetime object.
+            ingestion_date = datetime.datetime.strptime(
+                event["time"], "%Y-%m-%dT%H:%M:%SZ"
+            )
+            # Formats the datetime object to a string with the format "%Y%m%d".
+            ingestion_data_str = ingestion_date.strftime("%Y%m%d")
+            # Parses the newly formatted string back into a datetime
+            # object with the desired structure.
+            ingestion_date_object = datetime.datetime.strptime(
+                ingestion_data_str, "%Y%m%d"
+            )
+
+            status_params = {
+                "file_to_create_path": s3_filepath,
+                "status": Status.SUCCEEDED,
+                "job_definition": None,
+                "ingestion_date": ingestion_date_object,
+            }
+            session.add(models.StatusTracking(**status_params))
+            session.commit()
+            return {"statusCode": 200, "body": "Success"}
+
+    return {"statusCode": 400, "body": "Invalid data level"}
+
+
+def batch_event_handler(event, engine):
+    """Handler for Batch event
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+    engine : sqlalchemy.engine.Engine
+        Engine object to interact with the database.
+
+    Returns
+    -------
+    dict
+        HTTP response
+    """
+
+    if event["detail-type"] != "Batch Job State Change":
+        return {"statusCode": 400, "body": "Invalid event"}
```

Comment:
I think in one of your other PRs you added a helper function for all these return types. Do you want to re-use that here/add something similar since it looks like you're calling a lot of these?